### PR TITLE
Add ability to specify nginx access log format

### PR DESCRIPTION
**Pull request by rvadim**
Hi,
This PR add ability for customizing nginx access log format through router deployment annotations.

We need this ability to migrate from deis 1.x.

### CODE REVIEW

**Comment by deis-admin** [comment]
Thanks for the contribution! Please ensure your commits follow our [style guide](http://docs.deis.io/en/latest/contributing/standards/#commit-style-guide). This code will be tested once a Deis maintainer reviews it.

**Comment by mboersma** [comment]
Jenkins, add to whitelist.

**Comment by krancour** [comment]
I'd want @jchauncey to have final say on this. This might affect the metrics that are collected from the router.

**Comment by jchauncey** [comment]
Yes if you change the router log format in any way you will definitely jack up the router metrics we collect. So while I'm not opposed to this we should have a warning if you change it.

**Comment by rvadim** [comment]
So, any tasks for me here? Need to add warning to log?

**Comment by krancour** [comment]
@rvadim I think you'd need to add a warning to the documentation for this feature.

**Comment by rvadim** [comment]
Done, but get CI error in infrastructure.

File: README.md [hunk 1]
```diff
@@ -251,7 +251,7 @@ _Note that Kubernetes annotation maps are all of Go type `map[string]string`.  A
...
 | <a name="http2-enabled"></a>deis-router | deployment | [router.deis.io/nginx.http2Enabled](#http2-enabled) | `"true"` | Whether to enable HTTP2 for apps on the SSL ports. |
-| <a name="log-format"></a>deis-router | deployment | [router.deis.io/nginx.logFormat](#log-format) | `"[$time_iso8601] - $app_name - $remote_addr - $remote_user - $status - "$request" - $bytes_sent - "$http_referer" - "$http_user_agent" - "$server_name" - $upstream_addr - $http_host - $upstream_response_time - $request_time"` | Nginx access log format. |
+| <a name="log-format"></a>deis-router | deployment | [router.deis.io/nginx.logFormat](#log-format) | `"[$time_iso8601] - $app_name - $remote_addr - $remote_user - $status - "$request" - $bytes_sent - "$http_referer" - "$http_user_agent" - "$server_name" - $upstream_addr - $http_host - $upstream_response_time - $request_time"` | Nginx access log format. *Warning!* if you change this to non-default value, log parsing in logger subsystem will be broken. Use this parameter if you completely understand what you're doing. |
```
Comment (krancour) [inline README.md]: Formatting and grammar fix:

Change:

> *Warning!* if you change this to non-default value...

> **Warning:** If you change this to a non-default value...

Technical accuracy fix:

> log parsing in logger subsystem will be broken

I'd like @jchauncey to verify, but I believe this impacts the _monitoring_ subsystem; not logging.

**Comment by bacongobbler** [comment]
thanks @rvadim!

**Pull request merged**